prov/shm/src/smr_ep.c

```diff
@@ -826,6 +802,47 @@ static void smr_cleanup_epoll(struct smr_sock_info *sock_info)
 	ofi_epoll_close(sock_info->epollfd);
 }
 
+static void smr_srx_close(struct smr_ep *ep)
+{
+	struct smr_srx_ctx *srx;
+	struct smr_rx_entry *rx_entry;
+
+	srx = ep->srx->fid.context;
+	if (!srx)
+		return;
+
+	while (!dlist_empty(&srx->recv_queue.list)) {
```

Comment:
Do the entries on all these queues correspond to posted operations from the user?  Should this generate completions?

Reply:
Yes. Do all posted receives that weren't matched generate completions when closing the EP?

Reply:
I think only usNIC doesn't generate completions for posted receives when the EP is closed.  It's helpful, because otherwise, the user may need to track the buffers.  Not all apps want to die just because one of its peers goes away.  That's the use case we'll want to target.